NullPointerException in com.defold.iap.IapGooglePlay.onPurchasesUpdated

```
java.lang.RuntimeException: 
  at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args (LoadedApk.java:1656)
  at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run (Unknown Source:2)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8167)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
Caused by: java.lang.NullPointerException: 
  at com.defold.iap.IapGooglePlay.onPurchasesUpdated (IapGooglePlay.java:305)
  at com.android.billingclient.api.zze.onReceive
  at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args (LoadedApk.java:1646)

```

--- a/extension-iap/src/java/com/defold/iap/IapGooglePlay.java
+++ b/extension-iap/src/java/com/defold/iap/IapGooglePlay.java
@@ -362,9 +362,13 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
      */
     @Override
     public void onPurchasesUpdated(BillingResult billingResult, List<Purchase> purchases) {
-        if (billingResult.getResponseCode() == BillingResponseCode.OK && purchases != null) {
-            for (Purchase purchase : purchases) {
-                handlePurchase(purchase, this.purchaseListener);
+        if (billingResult.getResponseCode() == BillingResponseCode.OK) {
+            if (purchases != null && !purchases.isEmpty()) {
+                for (Purchase purchase : purchases) {
+                    if (purchase != null) {
+                        handlePurchase(purchase, this.purchaseListener);
+                    }
+                }
             }
         }
         else {
@@ -413,7 +417,12 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
                 @Override
                 public void onProductDetailsResponse(BillingResult billingResult, List<ProductDetails> productDetailsList) {
                     if (billingResult.getResponseCode() == BillingResponseCode.OK && (productDetailsList != null) && !productDetailsList.isEmpty()) {
-                        buyProduct(productDetailsList.get(0), token, purchaseListener);
+                        for (ProductDetails productDetails : productDetailsList) {
+                            if (productDetails != null) {
+                                buyProduct(productDetails, token, purchaseListener);
+                                break;
+                            }
+                        }
                     }
                     else {
                         Log.e(TAG, "Unable to get product details before buying: " + billingResult.getDebugMessage());
@@ -435,10 +444,12 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
 
             @Override
             public void onProductDetailsResponse(BillingResult billingResult, List<ProductDetails> productDetails) {
-                if (productDetails != null) {
+                if (productDetails != null && !productDetails.isEmpty()) {
                     // cache products (cache will be used to speed up buying)
                     for (ProductDetails pd : productDetails) {
-                        IapGooglePlay.this.products.put(pd.getProductId(), pd);
+                        if (pd != null) {
+                            IapGooglePlay.this.products.put(pd.getProductId(), pd);
+                        }
                     }
                     // add to list of all product details
                     allProductDetails.addAll(productDetails);
@@ -485,9 +496,11 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
             @Override
             public void onProductDetailsResponse(BillingResult billingResult, List<ProductDetails> productDetails) {
                 JSONArray a = new JSONArray();
-                if (billingResult.getResponseCode() == BillingResponseCode.OK) {
+                if ((billingResult.getResponseCode() == BillingResponseCode.OK) && (productDetails != null) && !productDetails.isEmpty()) {
                     for (ProductDetails pd : productDetails) {
-                        a.put(convertProductDetails(pd));
+                        if (pd != null) {
+                            a.put(convertProductDetails(pd));
+                        }
                     }
                 }
                 else {

--- a/game.project
+++ b/game.project
@@ -16,7 +16,7 @@ target_sdk_version = 29
 
 [project]
 title = extension-iap
-dependencies = https://github.com/andsve/dirtylarry/archive/master.zip
+dependencies#0 = https://github.com/andsve/dirtylarry/archive/master.zip
 
 [library]
 include_dirs = extension-iap

